### ISSUE TITLE
How do you set environment variables?

### ISSUE THREAD

**Issue by themire** [replicate/cog#1627]
There is a page listing out all the environment variables but no reference in the docs about cog.yaml about where and how these should be included. 

I've tried to figure it out from old PRs but couldn't find anything conclusive.

**Comment by zeke** [comment]
Thanks for reporting. Cog's support for env/secrets at build time and run time is not yet well documented.

Can you share a bit more detail about what you're trying to accomplish?

If you want to set environment variables when _running_ a model (not building), you can pass the `-e` flag to `cog predict` or `cog run`. See https://github.com/replicate/cog/pull/1253

If you want to inject secrets into your built image, see https://github.com/replicate/cog/blob/main/docs/private-package-registry.md

**Comment by themire** [comment]
Thanks. I am trying to trying to set the `WEBHOOK_AUTH_TOKEN` as documented [here](https://cog.run/environment/#webhook_auth_token).

**Comment by zeke** [comment]
I see. Are you running Cog on your own infrastructure and wanting to be able to send webhooks?

**Comment by themire** [comment]
Yes I am. The webhooks are all working fine. I just would like to be able to set this `WEBHOOK_AUTH_TOKEN` for extra security at my webhook endpoint.

**Comment by kgrv** [comment]
How can i set up environment variables to a prediction model if the model needs to communicate with external services (API endpoints, etc)?

**Comment by markphelps** [comment]
Closing this as stale since there hasn't been any activity in over a year. If you still need help with setting environment variables like WEBHOOK_AUTH_TOKEN, feel free to reopen or submit a PR to improve the documentation around environment variables and secrets.
